warehouse/query-core/src/test/java/datawave/query/LongRunningQueryTest.java

```diff
@@ -0,0 +1,140 @@
+        datawavePrincipal = new DatawavePrincipal((Collections.singleton(user)));
+        
+        TimeZone.setDefault(TimeZone.getTimeZone("GMT"));
+        
+        logic.setIncludeGroupingContext(true);
+        logic.setIncludeDataTypeAsField(true);
+        logic.setMarkingFunctions(new MarkingFunctions.Default());
+        logic.setMetadataHelperFactory(new MetadataHelperFactory());
+        logic.setDateIndexHelperFactory(new DateIndexHelperFactory());
+        logic.setResponseObjectFactory(new DefaultResponseObjectFactory());
+        
+        QueryTestTableHelper testTableHelper = new QueryTestTableHelper(LongRunningQueryTest.class.toString(), log);
+        recordWriter = new MockAccumuloRecordWriter();
+        testTableHelper.configureTables(recordWriter);
+        connector = testTableHelper.connector;
+        
+        // Load data for the test
+        VisibilityWiseGuysIngest.writeItAll(connector, VisibilityWiseGuysIngest.WhatKindaRange.DOCUMENT);
+        PrintUtility.printTable(connector, auths, TableName.SHARD);
+        PrintUtility.printTable(connector, auths, TableName.SHARD_INDEX);
+        PrintUtility.printTable(connector, auths, QueryTestTableHelper.MODEL_TABLE_NAME);
+    }
+    
+    /**
+     * A groupBy query is one type of query that is allowed to be "long running", so that type of query is used in this test.
+     *
+     * A long running query will return a ResultsPage with zero results if it has not completed within the query execution page timeout. This test expects 2
+     * pages, the first of which should have 0 results and be marked as PARTIAL. The second page should have 8 results and have a status of COMPLETE.
+     *
+     * @throws Exception
+     */
+    @Test
+    @Category(IntegrationTest.class)
+    public void testAllowLongRunningQueryWithShardQueryLogic() throws Exception {
+        
+        Map<String,String> extraParameters = new HashMap<>();
+        extraParameters.put("group.fields", "AGE,$GENDER");
+        extraParameters.put("group.fields.batch.size", "6");
+        
+        String queryStr = "UUID =~ '^[CS].*'";
+        Date startDate = format.parse("20091231");
+        Date endDate = format.parse("20150101");
+        QueryImpl query = new QueryImpl();
+        query.setQuery(queryStr);
+        query.setBeginDate(startDate);
+        query.setEndDate(endDate);
+        query.setQueryAuthorizations(auths.serialize());
+        query.setColumnVisibility("A&E&I");
+        query.setPagesize(Integer.MAX_VALUE);
+        query.setParameters(extraParameters);
+        query.setId(UUID.randomUUID());
+        
+        // this parameter is what makes the query long running. Failing to set this will let it default to 50 minutes
+        // (and not the 1000 milliseconds that it is set to) which will return only 1 page of 8 results, thereby failing this test.
+        logic.setQueryExecutionForPageTimeout(1000);
```

Comment:
I'm probably missing something.  Isn't there a chance this can execute so quickly that we'll still get results in the first page?

Reply:
Ha, yes, you're correct. In fact, that's the reason this test passes locally, but fails on github. I will put in a forced sleep or otherwise guarantee that the timeout is hit.